src/sage/matrix/matrix0.pyx

```diff
@@ -1920,6 +1922,21 @@ cdef class Matrix(sage.structure.element.Matrix):
             [ 0.333333333333333   66.6666666666667]
             [ -3.00000000000000 1.00000000000000e6]
 
+        Matrices with borders::
+
+            sage: M = matrix([[1,2,3], [4,5,6], [7,8,9]])
+            sage: M.subdivide(None, 2)
+            sage: print(M.str(unicode=True,
+            ....:             top_border=['ab', 'cde', 'f'],
+            ....:             bottom_border=['*', '', ''],
+            ....:             left_border=[1, 10, 100],
+            ....:             right_border=['', ' <', '']))
+                        ab cde   f
+                     1⎛  1   2│  3⎞
+                    10⎜  4   5│  6⎟ <
```

Comment:
wasn't sure about this one. i thought it would be too far away from the parenthesis

Reply:
i could make this another parameter, of course...

Reply:
It's ok like that.

However, it took me some time to understand the number of left spaces before the values of the left border. These spaces have nothing to do with the call to `rjust(left_width)` but are choices of the author in the presentation of the doctests. I suggest to remove the extra spaces to be consistent with what a user obtain in a terminal. It will also ease the work of future maintainers.

Reply:
Thanks, done in d45422e64ea